### PR TITLE
upgrade InfiniteGrid

### PR DESCRIPTION
**Pull request by sculove**
## Details

- When prepending, change the order of item
- When prepending, change the order of removed items
- remove croppedDistance property in 'prepend' event
- change global variable name from $global to $view
- fix position value to avoid bug

## Reference
#98, #99, #101, #102

## Preferred Reviewer
@kishu @netil

### CODE REVIEW

File: src/infiniteGrid.js [hunk 1]
```diff
@@ -58,18 +58,28 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document, "Outlayer"], function
 				item.getSize();
 			}
 			(item.isAppend == null) && (item.isAppend = true);
+			var outerHeight = parseInt(item.size.outerHeight, 10);
 			var y;
 			var shortColIndex;
 			var isAppend = item.isAppend;
 			var cols = isAppend ? this._appendCols : this._prependCols;
 			y = Math[isAppend ? "min" : "max"].apply(Math, cols);
```
Comment (kishu) [inline src/infiniteGrid.js]: ```var y = Math[isAppend ? "min" : "max"].apply(Math, cols);```
remove upside definition of ```var y'```

File: src/infiniteGrid.js [hunk 2]
```diff
@@ -338,11 +348,14 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document, "Outlayer"], function
...
 						 */
+						var croppedDistance = this.fit();
+						if (croppedDistance > 0) {
+							scrollTop -= croppedDistance;
+							this.$view.scrollTop(scrollTop);
+						}
 						this.trigger(this._prefix + EVENTS.prepend, {
-							scrollTop: scrollTop,
-							croppedDistance: this.fit()
+							scrollTop: scrollTop
```
Comment (kishu) [inline src/infiniteGrid.js]: I'm not sure pass ```scrollTop``` on

File: src/infiniteGrid.js (hunk 1 above)
Comment (sculove) [inline src/infiniteGrid.js]: I apply it

**Comment by kishu** [comment]
LGTM

File: src/infiniteGrid.js [hunk 3]
```diff
@@ -284,8 +293,8 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document, "Outlayer"], function
 			this._topElement = null;
 			this._bottomElement = null;
 			this._refreshViewport();
-			this.$global.on("resize", this._onResize);
-			this.$global.on("scroll", this._onScroll);
+			this.$view.on("resize", this._onResize);
+			this.$view.on("scroll", this._onScroll);
```
Comment (netil) [inline src/infiniteGrid.js]: chaining would be better
```js
this.$view.on("resize", this._onResize).on("scroll", this._onScroll);
```

**Pull request closed without merging**